lib/modules/backupAndRestore/providers/GoogleDrive.js

```diff
@@ -28,9 +34,42 @@ export class GoogleDrive extends Provider {
 			await Permissions.request(['https://accounts.google.com/o/oauth2/v2/auth', 'https://content.googleapis.com/drive/v3/*']);
 		});
 
+		// Re-check email address cache, in case it got invalidated due to google account logout
+		if (!this.getCachedEmailAddress()) {
+			try {
+				this.emailAddress = await this.verifyToken();
+				this.setCachedEmailAddress(this.emailAddress);
+			} catch (err) {
+				console.error(err);
+			}
+		}
+
 		return this;
 	}
 
+	getCachedEmailAddress() {
+		const cachedEmailAddress = localStorage.getItem(CACHED_GOOGLE_EMAIL_ADDRESS);
+		return cachedEmailAddress || '';
+	}
+
+	setCachedEmailAddress(emailAddress: string) {
+		localStorage.setItem(CACHED_GOOGLE_EMAIL_ADDRESS, JSON.stringify(emailAddress));
+	}
+
+	async verifyToken(): Promise<string> {
+		const decryptedToken = await ajax({
+			method: 'GET',
+			url: 'https://www.googleapis.com/oauth2/v3/tokeninfo',
+			query: { access_token: this.accessToken },
```

Comment:
Can access token be specified as a header, like in the other calls?

Reply:
Unfortunately, no, the Google API specifies that this access token should be sent as a query parameter. Ref.: https://developers.google.com/identity/protocols/OAuth2UserAgent#validate-access-token

I also tried to send this info in an `Authorization` header, but got a 400 Bad Request response:

```
curl -X POST \
  https://www.googleapis.com/oauth2/v3/tokeninfo \
  -H 'authorization: Bearer <access token...>' \
  -H 'cache-control: no-cache' \
  -H 'postman-token: c7c9317b-2089-e3a0-c4eb-dc076de14af6'
```

Response: 

```
Status: 400 Bad Request
{
    "error_description": "either access_token, id_token, or token_handle required"
}
```